src/lib.rs

```diff
@@ -69,6 +69,40 @@ pub mod virtual_dom;
 /// contain own initialization code.
 pub fn initialize() {
     stdweb::initialize();
+    js! {
+        var task = null;
+        var pool = [];
+        var routine = function() { };
+        var schedule_routine = function() {
```

Comment:
Just a tradition to pass `no-use-before-define` lint. Is it safe to take advantage of `var` behaviour when everything actually defines at top? Will it work in the future? I saw that `let` doesn't work this way. What do you think?

Reply:
Having two functions that call each other can get a little hairy. Is there a reason that `schedule_routine` is a different function? My immediate thoughts on this are that you could just inline that function and avoid the two functions referencing each other. Alternatively, the `schedule_routine` function can take `routine` as an argument, if you want it to be a general purpose utility to avoid code reuse. Another options is to pull the if block that calls schedule_routine into another funciton that is put into the setTimeout, so it would be something like:

```
...
task = setTimeout(() => {
    routine();
    if (pool.length > 0) {
        schedule_routine();
    }
});
...
```